internal/services/storage/storage_account_resource.go

```diff
@@ -233,10 +235,10 @@ func resourceStorageAccount() *pluginsdk.Resource {
 				ForceNew: true,
 			},
 
-			"allow_blob_public_access": {
+			allowPublicNestedItemsName: {
```

Comment:
we should add a TODO here to document this in 3.0

```suggestion
			// TODO: document this new field in 3.0
			allowPublicNestedItemsName: {
```